src/System.Private.Interop/src/Internal/Runtime/CompilerHelpers/RuntimeInteropData.CoreRT.cs

```diff
@@ -0,0 +1,204 @@
+                            closedStub = externalReferences.GetIntPtrFromIndex(entryParser.GetUnsigned());
+                            delegateCreationStub = externalReferences.GetIntPtrFromIndex(entryParser.GetUnsigned());
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        private unsafe bool TryGetStructData(RuntimeTypeHandle structTypeHandle, out ExternalReferencesTable externalReferences, out NativeParser entryParser)
+        {
+            int structHashcode = structTypeHandle.GetHashCode();
+            externalReferences = default(ExternalReferencesTable);
+            entryParser = default(NativeParser);
+            foreach (NativeFormatModuleInfo module in ModuleList.EnumerateModules())
+            {
+                NativeReader structMapReader;
+                if (TryGetNativeReaderForBlob(module, ReflectionMapBlob.StructMarshallingStubMap, out structMapReader))
+                {
+                    NativeParser structMapParser = new NativeParser(structMapReader, 0);
+                    NativeHashtable structHashtable = new NativeHashtable(structMapParser);
+
+                    externalReferences.InitializeCommonFixupsTable(module);
+
+                    var lookup = structHashtable.Lookup(structHashcode);
+                    while (!(entryParser = lookup.GetNext()).IsNull)
+                    {
+                        RuntimeTypeHandle foundStructType = externalReferences.GetRuntimeTypeHandleFromIndex(entryParser.GetUnsigned());
+                        if (foundStructType.Equals(structTypeHandle))
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+        private unsafe bool TryGetMarshallersForStruct(RuntimeTypeHandle structTypeHandle, out IntPtr marshalStub, out IntPtr unmarshalStub, out IntPtr destroyStub, out bool hasInvalidLayout, out int size)
+        {
+            marshalStub = IntPtr.Zero;
+            unmarshalStub = IntPtr.Zero;
+            destroyStub = IntPtr.Zero;
+            hasInvalidLayout = true;
+            size = 0;
+
+            ExternalReferencesTable externalReferences;
+            NativeParser entryParser;
+            if (TryGetStructData(structTypeHandle, out externalReferences, out entryParser))
+            {
+                marshalStub = externalReferences.GetIntPtrFromIndex(entryParser.GetUnsigned());
+                unmarshalStub = externalReferences.GetIntPtrFromIndex(entryParser.GetUnsigned());
+                destroyStub = externalReferences.GetIntPtrFromIndex(entryParser.GetUnsigned());
+                size = (int)entryParser.GetUnsigned();
+                uint mask = entryParser.GetUnsigned();
+                hasInvalidLayout = (mask & 0x1) == 1;
```

Comment:
We only need it here. The other place where it was computed was redundant, removed that.